src/projects/EnsureThat/EnumerableArg.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using EnsureThat.Annotations;
+using EnsureThat.Extensions;
+using JetBrains.Annotations;
+
+namespace EnsureThat
+{
+    /// <summary>
+    /// Ensures for <see cref="IEnumerable{T}"/>.
+    /// </summary>
+    /// <remarks>MULTIPLE ENUMERATION OF PASSED ENUMERABLE IS POSSIBLE.</remarks>
+    [SuppressMessage("ReSharper", "PossibleMultipleEnumeration")]
```

Comment:
Might be better to use https://www.jetbrains.com/help/resharper/Reference__Code_Annotation_Attributes.html#NoEnumerationAttribute on the IsNotNull to avoid this, rather than suppressing.

Reply:
(it's the opposite of InstantHandle, and tells R# that it NEVER enumerates the collection)

Reply:
"Indicates that IEnumerable, passed as parameter, is not enumerated"

That would be wrong. It is enumerated. The attribute that is added "SuppressMessage" is only for devs of Ensure.That. To get rid of squigglies in methods in this class where the enum is accessed twice.

Reply:
On IsNotNull - it's not enumerated for IsNotNull which is why you're getting squigglies :)

Reply:
It's already on the matching overload https://github.com/danielwertheim/Ensure.That/blob/master/src/projects/EnsureThat/AnyArg.cs#L10

And even if I remove: https://github.com/danielwertheim/Ensure.That/blob/master/src/projects/EnsureThat/EnumerableArg.cs#L24

It still complains as it's seen as used on:

https://github.com/danielwertheim/Ensure.That/blob/master/src/projects/EnsureThat/EnumerableArg.cs#L26

and

https://github.com/danielwertheim/Ensure.That/blob/master/src/projects/EnsureThat/EnumerableArg.cs#L32